### ISSUE TITLE
MA0042 shouldn't trigger on ADO.Net "using" declarations and statements

### ISSUE THREAD

**Issue by robv8r** [meziantou/Meziantou.Analyzer#1117]
### Version of the Meziantou.Analyzer NuGet package

3.0.61

### Rule Identifier

MA0042

### Target Framework

net10.0

### C# Language version

C# 14.0

### Description

The Analyzer raises MA0042 for `using` declarations and statements for ADO.Net.

### Reproduction Steps

Minimal code:

```c#
// MA0042 "Prefer using 'await using'
using var connection = new SqlConnection(this.ConnectionString);
await connection.OpenAsync(token).ConfigureAwait(false);

// MA0042 "Prefer using 'await using'
using var command = new SqlCommand("dbo.ExampleStoredProcecure", connection)
{
    CommandType = CommandType.StoredProcedure,
};

// MA0042 "Prefer using 'await using'
using var reader = await command.ExecuteReaderAsync(token).ConfigureAwait(false);
```

### Other information

After accepting the Analyzer's "fix," the code does not compile.

This applies to both `using` declarations and `using statements`.

For example:

```c#
// MA0042 "Prefer using 'await using'
using (var connection = new SqlConnection(this.ConnectionString))
{
}
```

**Comment by meziantou** [comment]
Can you clarify what doesn't work after applying the fix? On my side:

````
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net10.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

	<ItemGroup>
		<PackageReference Include="Meziantou.Analyzer" Version="3.0.61">
			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
			<PrivateAssets>all</PrivateAssets>
		</PackageReference>
		<PackageReference Include="Microsoft.Data.SqlClient" Version="7.0.1" />
	</ItemGroup>

</Project>
````

The fixed code is:

```c#
// MA0042 "Prefer using 'await using'
await using var connection = new SqlConnection("");
await connection.OpenAsync(CancellationToken.None).ConfigureAwait(false);

// MA0042 "Prefer using 'await using'
await using var command = new SqlCommand("dbo.ExampleStoredProcecure", connection)
{
    CommandType = CommandType.StoredProcedure,
};

// MA0042 "Prefer using 'await using'
await using var reader = await command.ExecuteReaderAsync(CancellationToken.None).ConfigureAwait(false);
```

Can you provide the original code and relevant NuGet packages, and the fixed code? It will help reproducing the error.

**Comment by robv8r** [comment]
Thank you for the quick response.  I believe my original issue was caused by user-error.  However, I think I've discovered a different issue that can cause `UseConfigureAwaitFixer` to crash.

I'm using Visual Studio 2026, version 18.5.1.

When using the above fix, I get different warnings:

```c#
// MA0004 - Use Task.ConfigureAwait(false) if the current synchronization context is not needed
await using var connection = new SqlConnection("");
await connection.OpenAsync(CancellationToken.None).ConfigureAwait(false);

// MA0004 - Use Task.ConfigureAwait(false) if the current synchronization context is not needed
await using var command = new SqlCommand("dbo.ExampleStoredProcecure", connection)
{
    CommandType = CommandType.StoredProcedure,
};

// MA0004 - Use Task.ConfigureAwait(false) if the current synchronization context is not needed
await using var reader = await command.ExecuteReaderAsync(CancellationToken.None).ConfigureAwait(false);
```

When I apply the `MA0004` fix to the `connection`, its declaration is moved outside a new `using` statement.  Attempting to fix the remaining `MA0004` warnings causes the `UseConfigureAwaitFixer` to crash (see stack trace below).

```c#
var connection = new SqlConnection("");

await using (connection.ConfigureAwait(false))
{
    await connection.OpenAsync(CancellationToken.None).ConfigureAwait(false);

    // MA0004 - Use Task.ConfigureAwait(false) if the current SynchronizationContext is not needed.
    // Applying fix crashes UseConfigureAwaitFixer
    await using var command = new SqlCommand("dbo.ExampleStoredProcecure", connection)
    {
        CommandType = CommandType.StoredProcedure,
    };

    // MA0004 - Use Task.ConfigureAwait(false) if the current SynchronizationContext is not needed.
    // Applying fix crashes UseConfigureAwaitFixer
    await using var reader = await command.ExecuteReaderAsync(CancellationToken.None).ConfigureAwait(false);
}
```

Stack trace from `UseConfigureAwaitFixer` when attempting to fix remaining `MA0004` warnings.
```
System.InvalidCastException : Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.VariableDeclaratorSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax'.
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGeneratorInternal.MemberAccessExpressionWorker(SyntaxNode expression,SyntaxNode simpleName)
   at Microsoft.CodeAnalysis.Editing.SyntaxGeneratorInternal.MemberAccessExpression(SyntaxNode expression,SyntaxNode memberName)
   at Meziantou.Analyzer.Rules.UseConfigureAwaitFixer.<AddConfigureAwait>g__AppendConfigureAwait|4_1(SyntaxNode expressionSyntax,<>c__DisplayClass4_0& )
   at async Meziantou.Analyzer.Rules.UseConfigureAwaitFixer.AddConfigureAwait(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputePreviewOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.DocumentChangeAction.ComputePreviewOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.EditorSuggestedAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.EditorSuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.EditorSuggestedActionWithNestedFlavors.<>c__DisplayClass16_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

If I apply the fixes from bottom-to-top, the `UseConfigureAwaitFixer` does not crash.

```c#
var connection = new SqlConnection("");
await using (connection.ConfigureAwait(false))
{
    await connection.OpenAsync(CancellationToken.None).ConfigureAwait(false);

    var command = new SqlCommand("dbo.ExampleStoredProcedure", connection)
    {
        CommandType = CommandType.StoredProcedure,
    };

    await using (command.ConfigureAwait(false))
    {
        var reader = await command.ExecuteReaderAsync(CancellationToken.None).ConfigureAwait(false);

        await using (reader.ConfigureAwait(false))
        {
        }
    }
}
```
